src/main/java/com/nylas/RoomResource.java

```diff
@@ -1,10 +1,14 @@
 package com.nylas;
 
-public class RoomResource {
+public class RoomResource extends AccountOwnedModel implements JsonObject {
```

Comment:
The RoomResource does not seem to be a first-class RestfulModel or AccountOwnedModel because it has no `id` or `account_id` fields.  That is why it currently is just accessed through the Events.roomResources method.  I'm not sure that should change here.  At least without understanding why they don't have ids.  I'm concerned that making RoomResource's into RestulModel objects with a RestfulDAO could lead to future mistakes by developers or users of the Java SDK that assume they are just like all the others.

Reply:
I don't think `RoomResource` should be accessed from the `Events.roomResources` method anymore as Room Resources are split off into its own endpoint and has its own section in the documentation decoupled from Events. In the other SDKs it's treated as it's own entity as well so in terms of consistency, the Java SDK would be the only one that ties Events to Room Resources. From my testing, the way it's set up right now the only method a user can call using `account.roomResources()`is `.list()` which is currently the only operation available via the API. Alternatively what we can do is make a base model without any id field that `RestfulModel` extends from, allowing us to extend `RoomResources` directly from that new model. The RoomResource lacking an `id` and `account_id` is not the prettiest I agree but I don't see it being the biggest of issues given that all you can do is list RoomResources at the moment. If there was POST/PUT/DELETE involved I would be more hesitant.

Reply:
I think it's reasonable to move RoomResources to its own DAO to reflect the organization of the documentation and consistency with the other SDKs.  But we can do that without making it a RestfulModel or the DAO a RestfulDAO.  Basically, copy Events.roomResrouces over to RoomResources.list (and deprecate Events.roomResources as I don't think we should support both).  In the future, if RoomResources become a more first class object with ids, then we can change the modeling to match.

Reply:
New commit reflects the feedback in your previous comment